pkg/apis/monitoring/v1/types.go

```diff
@@ -615,27 +637,66 @@ type PodMetricsEndpoint struct {
 // More info: https://prometheus.io/docs/operating/configuration/#endpoints
 // +k8s:openapi-gen=true
 type BasicAuth struct {
-	// The secret that contains the username for authenticate
+	// The secret in the service monitor namespace that contains the username
+	// for authentication.
 	Username v1.SecretKeySelector `json:"username,omitempty"`
-	// The secret that contains the password for authenticate
+	// The secret in the service monitor namespace that contains the password
+	// for authentication.
 	Password v1.SecretKeySelector `json:"password,omitempty"`
 }
 
 // TLSConfig specifies TLS configuration parameters.
 // +k8s:openapi-gen=true
 type TLSConfig struct {
-	// The CA cert to use for the targets.
+	// Path to the CA cert in the Prometheus container to use for the targets.
 	CAFile string `json:"caFile,omitempty"`
-	// The client cert file for the targets.
+	// Secret containing the CA cert to use for the targets.
+	CASecret *v1.SecretKeySelector `json:"caSecret,omitempty"`
```

Comment:
At least for our own usage we'll already require this to be a configmap instead. I think we'll need to for everything where we use a secretkeyselector right now, instead do either configmapkeyselector _or_ secretkeyselector.